.gitignore

```diff
@@ -12,4 +12,5 @@ docs/*
 doc/*
 bin/querypath-200x333.png
 test/fakepear
-vendor/
\ No newline at end of file
+vendor/
+composer.lock
```

Comment:
@joelpittet the removal of the lock file is addressed in the PR description

> Removes composer.lock file to ensure the correct PHPUnit is installed in GitHub Actions. [Since this is a library this is generally the accepted practice](https://stackoverflow.com/a/24247443).

Reply:
Thanks for referencing where it was mentioned. It seems like it has no affect downstream and I'd assume the opposite reading the official docs, if you want Github Actions to use a consistent version of PHPUnit you'd want to keep it.
https://getcomposer.org/doc/02-libraries.md#lock-file

Reply:
> if you want Github Actions to use a consistent version of PHPUnit you'd want to keep it.

That's exactly what we don't want to happen. If there was a fixed version of PHPUnit installed, you would be unable to automate the unit tests from PHP 7.1 all the way up until PHP 8.1. That's because there is no single version of PHPUnit that is compatible with the range of PHP versions currently supported by this library. This is why this PR replaces the dependency `phpunit/phpunit` with `yoast/phpunit-polyfills`, which has been specifically designed for this use case.

Reply:
Ah you have much more domain knowledge on this than I do. Thanks for sharing your experience